_posts/2016-10-24-deploy-jekyll-with-travis.md

```diff
@@ -0,0 +1,232 @@
+---
+layout: post
+title: Deploying Jekyll to Github Pages with Travis and Gulp
+date: 2016-10-24
+author: Anne Tomasevich
+tags: jekyll travis
+summary: |
+  A step-by-step guide to how we deploy our gulpified Jekyll site to GitHub
+  Pages using Travis, opening the door for us to use Jekyll Plugins.
+---
+
+Since our company's inception, we at Savas Labs have used [Jekyll](https://jekyllrb.com/) to build our company website and [GitHub Pages](https://pages.github.com/) to host it with great success. Jekyll is a powerful tool out of the box, and it's hard to imagine a simpler hosting strategy than GitHub Pages, which only requires a [specifically named branch](https://help.github.com/articles/configuring-a-publishing-source-for-github-pages/) and a [single settings update](https://help.github.com/articles/configuring-a-publishing-source-for-github-pages/#enabling-github-pages-to-publish-your-site-from-master-or-gh-pages) to get your site live for free.
+
+One well-known (and completely understandable) limitation of this workflow: any Jekyll plugins not [whitelisted by GitHub Pages](https://pages.github.com/versions/) will be disabled when GitHub Pages builds the site, since `jekyll build` is run with the `--safe` tag. Many solutions exist to get around this, and when we decided to start using Jekyll plugins on our site we opted to use [Travis CI](https://travis-ci.org/) to build our site and push the compiled site to our `master` branch, to then be launched via GitHub Pages. To complicate matters, we recently starting building our site with gulp, which introduced another layer of dependencies to the mix.
+
+## Basic Workflow
+
+Our site was already using Travis CI to run our test suite on pull requests and
+merges to the `master` branch. You can learn about setting up Travis in their
+[docs](https://docs.travis-ci.com/).
+
+Previously, Travis would build the site and run our tests, then we would merge
+branches into `master`, triggering GitHub Pages to rebuild and deploy our site.
+But we want to build our site before it gets to GitHub Pages to ensure our
+Jekyll Plugins are included. In our new workflow:
+
+1. We'll create a new default branch so the `master` branch can be cleared
+out and used by Travis.
+2. When this new branch is updated (i.e. a pull request is merged), Travis will
+build the site.
+3. Using a GitHub token for authorization, Travis will commit the `_site`
+directory to the `master` branch, then push the `master` branch.
+4. This will trigger GitHub Pages to deploy our pre-built site.
+
```

Comment:
If it's not too much work, I think it could be helpful to use a diagram illustrating the before and after workflows (like the Sketch ones you used in your last post) -- I think the visual could add a lot by emphasizing the differences and how things work.